doc/source/development/roadmap.rst

```diff
@@ -0,0 +1,145 @@
+especially for wide DataFrames. However, the BlockManager substantially increases the
+complexity and maintenance burden of pandas'.
+
+By replacing the BlockManager we hope to achieve
+
+* Substantially simpler code
+* Easier extensibility with new logical types
+* Better user control over memory use and layout
+* Improved microperformance
+* Option to provide a C / Cython API to pandas' internals
+
+See `these design documents <https://dev.pandas.io/pandas2/internal-architecture.html#removal-of-blockmanager-new-dataframe-internals>`__
+for more.
+
+Decoupling of Indexing and Internals
+------------------------------------
+
+The code for getting and setting values in pandas' data structures needs refactoring.
+In particular, a clear separation must be made between code that
+converts keys (e.g., the argument to ``DataFrame.loc``) to positions from code that uses
+uses these positions to get or set values. This is related to the proposed BlockManager rewrite.
+Currently, the BlockManager sometimes uses label-based, rather than position-based, indexing.
+We propose that it should only work with positional indexing, and the translation of keys
+to positions should be entirely done at a higher level.
+
+Indexing is a complicated API with many subtleties. This refactor will require care
+and attention. More details are discussed at
+https://github.com/pandas-dev/pandas/wiki/(Tentative)-rules-for-restructuring-indexing-code
+
+Weighted Operations
+-------------------
+
+In many fields, sample weights are necessary to correctly estimate population
+statistics. We'd like to support weighted operations (like `mean`, `sum`, `std`,
+etc.), possibly with an API similar to `DataFrame.groupby`.
+
+See https://github.com/pandas-dev/pandas/issues/10030 for more.
+
+Documentation Improvements
+--------------------------
+
+We'd like to improve the content, structure, and presentation of pandas documentation.
+Some specific goals include
+
+* Overhaul the HTML theme with a modern, responsive design.
+* Improve the "Getting Started" documentation, designing and writing learning paths
+  for users different backgrounds (e.g. brand new to programming, familiar with
+  other languages like R, already familiar with Python).
+* Improve the overall organization of the documentation and specific subsections
+  of the documentation to make navigation and finding content easier.
+
+Package Docstring Validation
+----------------------------
+
+To improve the quality and consistency of pandas docstrings, we've developed
+tooling to check docstrings in a variety of ways.
+https://github.com/pandas-dev/pandas/blob/master/scripts/validate_docstrings.py
+contains the checks.
+
+Like many other projects, pandas uses the
+`numpydoc <https://numpydoc.readthedocs.io/en/latest/>`__ style for writing
+docstrings. With the collaboration of the numpydoc maintainers, we'd like to
+move the checks to a package other than pandas so that other projects can easily
+use them as well.
```

Comment:
I would add a section on numba integration, something like.

```
Numba Integration
----------------------

Rewrite of much of the computation and aggregation code using numba. This will provide a) a much lower barrier to entry for newcomers and you now debug python code, b) simpler and shared code across different areas of pandas, possibly including groupby & windowing operations, potentially, c) potential improved performance as operations now can be JITed

Reply:
Side-note on 

> potential improved performance as operations now can be JITed

Do we expect to get improved performance?  I wouldn't expect e.g. `Series.sum()` to improve. IMO the best-case option here is accepting Numba-jitted user defined functions in a `rolling().apply(func)`, which would be very neat to see.

Reply:
Check out some of the work in https://github.com/shoyer/numbagg - it's originally computations written in numba for multi-dimensional arrays, but would work just as well for 1-2D arrays. 

I copied the pandas implementation of `ewma` (with attribution :) ) so we could have that functionality in xarray

CC @shoyer